src/bin/bat/app.rs

```diff
@@ -257,7 +257,7 @@ impl App {
         let files: Option<Vec<&OsStr>> = self.matches.values_of_os("FILE").map(|vs| vs.collect());
 
         if files.is_none() {
-            let input = Input::stdin().with_name(filenames_or_none.next().unwrap_or(None));
+            let input = Input::stdin().as_file(filenames_or_none.next().unwrap_or(None));
```

Comment:
I liked the previous name better. Without looking at the definition, it's not clear what `as_file` does and what the argument should be. Maybe `.as_file_with_name(…)` or `.kind("File").with_name(…)`? Or `.as_file().with_name(…)`?

Reply:
I'll see if `.as_file()` could work.

Reply:
I ended up just removing `as_file` entirely, since we only ever really use it for pretending a STDIN input is a file. In its place, I added `Input::stdin_as_file(Option)`, which describes the purpose a bit clearer.

In case we ever use reader inputs along with `--file-name` in the `bat` application itself, we can always revert it back to being a separate function, but for now, I think this is most straightforward way to implement it.

It also fixes the `File: STDIN` issue mentioned below.